### ISSUE TITLE
Allow to specify request headers for a WebSocket client

### ISSUE THREAD

**Issue by SirNickolas** [rejectedsoftware/vibe.d#1658]
Thanks for this great library!

I'd like to use it in a desktop application for communication with a web server via WebSockets. However, HTTP headers for the handshake are hardcoded, and there is no way to modify them. More precisely, I cannot add cookies which are essential for authentication.

Maybe an additional callback to `connectWebSocket` is needed? Or any other means. I'm new to vibe.d and don't know its best practices.

**Comment by s-ludwig** [comment]
You are right, this was an oversight when the WebSocket client support was added recently. I'm currently not sure if just adding an additional callback parameter is the best solution, or if it makes sense to introduce a `WebSocketSettings` type that contains such a callback as a delegate field. Maybe it makes sense to also make certain settings configurable, such as keep-alive behavior.
